core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -1190,6 +1192,9 @@ class PartIdCollector extends SimpleCollector {
   override def collect(doc: Int): Unit = {
     if (partIdDv.advanceExact(doc)) {
       result += partIdDv.longValue().toInt
+      if (result.length >= limit) {
```

Comment:
I think we should move these checks right when the ``collect`` call is made.  While the document isn't very clear it will terminate the further collection  on the segment but that means there will be one record collected every other segment before we stop collecting in that segment so we will essentially collect ``limit + (segments - 1)`` results. \